### PR TITLE
Allow to get the target and the certificate path from X509CertPath 

### PR DESCRIPTION
**Pull request by AlexITC**
Signed-off-by: Alexis Hernandez <alexis22229@gmail.com>

### CODE REVIEW

**Comment by sophokles73** [comment]
IMHO it would be sufficient to add a *getCertPath* method to `org.eclipse.californium.scandium.auth.X509CertPath`. No need to introduce an additional field all along the stack ...

**Comment by AlexITC** [comment]
@sophokles73 you are right, its quite simpler in this way, thanks!

File: scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java [hunk 1]
```diff
@@ -97,4 +97,18 @@ public static X509CertPath fromBytes(final byte[] encodedPath) {
 	public String getName() {
 		return target.getSubjectX500Principal().getName();
 	}
+
+	/**
+	 * Gets this certificate path.
+	 *
+	 * @return The path.
+	 */
+	public CertPath getPath() { return path; }
```
Comment (sophokles73) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java]: please put the method body on its own line (like in the the rest of the class)

File: scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java [hunk 2]
```diff
@@ -97,4 +97,18 @@ public static X509CertPath fromBytes(final byte[] encodedPath) {
...
+
+	/**
+	 * Gets the asserted identity of this certificate path.
+	 *
+	 * @return The subject.
```
Comment (sophokles73) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java]: this is not the subject but the whole certificate

File: scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java [hunk 3]
```diff
@@ -664,7 +664,7 @@ public Principal getPeerIdentity() {
 	/**
 	 * Sets the authenticated peer's identity.
 	 * 
-	 * @param the identity
+	 * @param peerIdentity the identity
```
Comment (sophokles73) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java]: this doesn't seem to be related to the other changes.
next time, please create a separate PR for it

File: scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java (hunk 1 above)
Comment (AlexITC) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java]: the method body is already in the same line as the method signature, isn't it?

Comment (sophokles73) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java]: what I mean is:
````
public CertPath getPath() {
  return path;
}
````

Comment (AlexITC) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/auth/X509CertPath.java]: sure, I placed it this way following the style of `getName`, I'll change it soon, thanks.

**Comment by AlexITC** [comment]
@sophokles73 I just noticed the `getName` method was not compacted, Intellij was rendering it in as compacted, sorry for that, code was updated. Thanks!
